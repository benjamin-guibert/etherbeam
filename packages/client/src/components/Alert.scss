@use '../styles/resources/sizes';

.my-alert {
  margin: sizes.$space-s 0px;
  padding: sizes.$space-s;
  border: sizes.$border solid;
  border-radius: sizes.$border-radius;
}
