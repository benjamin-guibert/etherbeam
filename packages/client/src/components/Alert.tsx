import React, { ReactElement, ReactNode } from 'react'
import './Alert.scss'

type AlertTypeProp = 'neutral' | 'primary' | 'secondary' | 'success' | 'danger'

interface AlertProps {
  type?: AlertTypeProp
  children?: ReactNode
}

const Alert = ({ type = 'neutral', children }: AlertProps): ReactElement => {
  return <div className={`my-alert my-color-${type}`}>{children}</div>
}

export default Alert
