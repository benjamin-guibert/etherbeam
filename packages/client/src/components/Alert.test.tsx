import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Alert from './Alert'

describe('<Alert />', () => {
  const content = 'Alert.'
  let component: ShallowWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = shallow(<Alert>{content}</Alert>)
    })

    it('should render', () => {
      const div = component.find('div').at(0)
      expect(div.hasClass('my-alert')).toBeTruthy()
      expect(div.hasClass('my-color-neutral')).toBeTruthy()
      expect(div.prop('children')).toBe(content)
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('Type', () => {
    beforeAll(() => {
      component = shallow(<Alert type="primary">{content}</Alert>)
    })

    it('should render', () => {
      const div = component.find('div').at(0)
      expect(div.hasClass('my-alert')).toBeTruthy()
      expect(div.hasClass('my-color-primary')).toBeTruthy()
      expect(div.prop('children')).toBe(content)
    })
  })
})
