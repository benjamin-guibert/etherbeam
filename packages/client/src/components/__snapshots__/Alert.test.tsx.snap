// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> Default should match snapshot 1`] = `
<div
  class="my-alert my-color-neutral"
>
  Alert.
</div>
`;
